validphys2/src/validphys/pdfplots.py

```diff
@@ -53,8 +53,9 @@ def __init__(self, pdfs, xplotting_grids, xscale, normalize_to, ymin, ymax):
         self.ymax = ymax
 
     def setup_flavour(self, flstate):
-        pass
-
+        flstate.handles=[]
+        flstate.labels=[]
+        flstate.hatchit=plotutils.hatch_iter()
```

Comment:
Why are we moving these things, which were specific to the bands pdfs here?

Reply:
Because `MixMatchPDFPlotter` inherits from both  `ReplicaPDFPlotter` and `BandPDFPlotter`, so the way  `ReplicaPDFPlotter` deals with labels and handles has also been changed to be the same.

Reply:
But this still uses the flstate handles and labels from `BandPDFPlotter`. The issue here is that `ReplicaPDFPlotter` has different instances for the handles and labels.

Reply:
Would most things work by inheriting from BandPDFPlotter first?

Reply:
Sure, all BandPDFPlotter and ReplicaPDFPlotter do is add a function on PDFPlotter to draw the figures and add the labels, where for ReplicaPDFPlotter the default matplotlib legend is used. 

What I did so far is move the label generation from BandPDFPlotter to PDFPlotter and add the labels to flstate also for ReplicaPDFPlotter. Instead of moving the label generation to PDFPlotter it is indeed also possible to make ReplicaPDFPlotter inherit from BandPDFPlotter but in my mind that organisation makes less sense.

Reply:
@Zaharid what would be your preferred solution? Feel free to change this PR yourself if that's easier than explaining.